### ISSUE TITLE
compiler.cpp includes codegencontext.cpp

### ISSUE THREAD

**Issue by rafalcieslak** [rafalcieslak/CCoscope#13]
In the header of `compiler.cpp`:

```
#include "codegencontext.h"
#include "codegencontext.cpp" // gcc needs it for some reason...
```

Why exactly is the `.cpp` file included? That doesn't seem right. In fact, it causes `CodegenContext::CodegenContext(...)` to be compiled twice, into `compiler.o` and then `codegencontext.o`, and thus the linker fails, as there are multiple definitions of that constructor available. Removing that line fixes the issue i.e. the sources compile successfully.

Can we remove that inclusion? Why was it needed in the first place?

**Comment by rafalcieslak** [comment]
Oh maybe you forgot to re-run `cmake ..` after creating `codegencontext.cpp` file, so the dependencies were not refreshed and the new file wasn't compiled at all.

**Comment by rafalcieslak** [comment]
I assumed this is the case and fixed this in 72fbc69. If I'm wrong, please reopen this issue.

**Comment by socumbersome** [comment]
Right, I forgot about `cmake ..` ;)
